tailcfg/tailcfg.go

```diff
@@ -1369,3 +1373,59 @@ type SetDNSRequest struct {
 	// Value is the value to add.
 	Value string
 }
+
+// SSHPolicy is the policy for how to handle incoming SSH connections
+// over Tailscale.
+type SSHPolicy struct {
+	// Rules are the rules to process for an incoming SSH
+	// connection. The first matching rule takes its action and
+	// stops processing further rules.
+	Rules []*SSHRule
+}
+
+// An SSH rule is a match predicate and associated action for an incoming SSH connection.
+type SSHRule struct {
+	// Principal matches an incoming connection. If the connection
+	// matches anything in this list and also matches SSHUser,
+	// then Action is applied.
+	Principal []*SSHPrincipal `json:"principal"`
+
+	// SSHUser are the SSH users that this rule matches.
+	// It is a map from either ssh-user|"*" => local-user.
+	// The map must contain a key for either ssh-user or "*" to
+	// match. If it does, the map entry's value is the actual
+	// user that's logged in.
+	SSHUser map[string]string `json:"sshUser"`
+
+	// Action is the outcome to task.
+	// A nil or invalid action means to deny.
+	Action *SSHAction `json:"action"`
+}
+
+// SSHPrincipal is either a particular node or a user on any node.
```

Comment:
What about tags? Do you intend on solving that by using `StableNodeID`?

Reply:
Yes, we can send the whole set down.